README.md

```diff
@@ -1,13 +1,67 @@
 # Moxie
 [![Build Status](https://travis-ci.org/DarthStrom/Moxie.svg?branch=master)](https://travis-ci.org/DarthStrom/Moxie)
-
 A tasty mocking library for Swift
 
 ![Moxie](moxie.jpeg)
 
+## Creating mocks
+All mock objects must conform to the Mock protocol and have an instance of Moxie.
+
+```swift
+import Moxie
+
+class MockClass: Mock {
+    var moxie = Moxie()
+}
+```
+
+Additionally, all mock objects must be of the same type as the object being mocked. This can be accomplished via extension, or adoption of the mocked object's protocol. If the object being mocked is not a protocol type (such as a struct), then it cannot be extended and so it must be mocked via protocol adoption.
+
+###  Extension
```

Comment:
Should we say Inheritance here?  I think extension generally means `extend ExampleClass {` etc.